Sources/Basic/PathShims.swift

```diff
@@ -17,22 +17,11 @@
  while making it fairly easy to find those calls later.
 */
 
-import SPMLibc
 import Foundation
 
 /// Returns the "real path" corresponding to `path` by resolving any symbolic links.
 public func resolveSymlinks(_ path: AbsolutePath) -> AbsolutePath {
-    let pathStr = path.pathString
-
-    // FIXME: We can't use FileManager's destinationOfSymbolicLink because
-    // that implements readlink and not realpath.
-    if let resultPtr = SPMLibc.realpath(pathStr, nil) {
-        let result = String(cString: resultPtr)
-        // FIXME: We should measure if it's really more efficient to compare the strings first.
-        return result == pathStr ? path : AbsolutePath(result)
-    }
-
-    return path
+    return AbsolutePath(path.pathString.resolvingSymlinksInPath)
```

Comment:
I understand and support the motivation here, but I have some concerns about performance.  For example, I think that the swift-corelibs-foundation implementation implements this by creating multiple URLs and breaking it into components.  Would it be possible to conditionally keep the optimized implementation and use the more generic one only for platforms that don't support `realpath()`?

We should of course measure to see whether there's really a difference, but shouldn't make this change unless we know it doesn't slow down the current platforms.

Reply:
Just a curious onlooker here, but this seems like a strange criticism: the Foundation API does exactly what this one is supposed to do, and if there are more optimal ways to get the result, that should be a Foundation-level optimisation which would benefit everybody.